src/components/Map/map.js

```diff
@@ -415,10 +461,33 @@ class Map extends Component {
       .setData({ type: "FeatureCollection", features: circles });
   };
 
-  updateZoom = distance => {
+  zoomToDistance = distance => {
+    let resolution = window.screen.height;
+    let latitude = this.props.search.coordinates[1];
+    let milesPerPixel = (distance * 8) / resolution;
+    return (
+        Math.log2(
+            (24901 * Math.cos((latitude * Math.PI) / 180)) / milesPerPixel
+        ) - 8
+    );
+  };
+
+  newSelection = (prevProps) => {
```

Comment:
Thinking about the case where you deselect all provider types or switch to the saved tab with unsaved, highlighted providers. This is more of a brain dump, happy to help flesh out details:

This method returns all the provider ID's that we need to add markers for. Similarly, we can have a method that returns all provider ID's that we need to remove existing markers for.

if h is set of highlighted provider id's, and p is the set of visible provider id's, and s is the set of provider id's we have markers for, then the markers we need to add are intersect(h, p) - s, and the markers we need to remove are s - intersect(h, p). intersect(h, p) equals currIds in this method. s equals this.selectionMarkers.map(m => m.providerId). So the id's we need to add markers for is currIds.filter(id => !s.includes(id)) and the id's we need to remove markers for is s.filter(id => !currIds.includes(id)).

So I don't think we actually need prevProps for this. You could also combine these into one method that returns {idsToAdd: [], idsToRemove:[]} and then also combine the removeDeselectedMarkers and markRecentSelection methods into an updateProviderMarkers method.